src/caret_analyze/plot/histogram/histogram_plot_factory.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2021 Research Institute of Systems Planning, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+
+from caret_analyze.record import Frequency, Latency, Period
+
+from .histogram_plot import HistogramPlot
+from ..visualize_lib import VisualizeLibInterface
+from ...common import type_check_decorator
+from ...exceptions import UnsupportedTypeError
+from ...runtime import CallbackBase, Communication
+
+HistogramPlotTypes = CallbackBase | Communication
+
+
+class HistogramPlotFactory:
+    """Factory class to create an instance of TimeSeriesPlot."""
+
+    @staticmethod
+    @type_check_decorator
+    def create_instance(
+        target_objects: Sequence[HistogramPlotTypes],
+        metrics: str,
+        visualize_lib: VisualizeLibInterface
+    ) -> HistogramPlot:
+        """
+        Create an instance of TimeSeriesPlot.
+
+        Parameters
+        ----------
+        target_objects : Sequence[TimeSeriesPlotTypes]
+            TimeSeriesPlotTypes = CallbackBase | Communication | (Publisher | Subscription)
+        metrics : str
+            Metrics for timeseries data.
+            supported metrics: [frequency/latency/period]
+        visualize_lib : VisualizeLibInterface
+            Instance of VisualizeLibInterface used for visualization.
+
+        Returns
+        -------
+        TimeSeriesPlot
+
+        Raises
+        ------
+        UnsupportedTypeError
+            Argument metrics is not "frequency", "latency", or "period".
```

Comment:
Update doc_string from the one for Timeseries to the one for Histogram.

There are several other similar mistakes.